Lib/shutil.py

```diff
@@ -635,82 +635,59 @@ def onerror(err):
         onexc(os.rmdir, path, err)
 
 # Version using fd-based APIs to protect against races
-def _rmtree_safe_fd(topfd, path, onexc):
+def _rmtree_safe_fd(stack, onexc):
+    func, dirfd, path, orig_entry = stack.pop()
```

Comment:
I'm not in love with `orig_entry` as it's unclear what this is without diving into the code to find what `orig_entry` actually is, when it can become non-None (in line 675), and why it can be None (initially, in line 744; and then in 665).

Reply:
We use it to save an `os.lstat()` when walking subdirectories. Perhaps I could stick a hefty comment above this line explaining the 4 values?